classes/class.ilMumieTaskGradeListGUI.php

```diff
@@ -0,0 +1,207 @@
+                        $this->tpl->setVariable("VAL_GRADE", round($xGrade->result->score->raw * 100));
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'user_id', $user_id);
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'newGrade', round($xGrade->result->score->raw * 100));
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'timestamp', strtotime($xGrade->timestamp));
+
+                        $this->tpl->setVariable("LINK_NAME", $this->ctrl->getLinkTarget($parentObj, 'displayGradeList'));
+
+                        $this->tpl->setVariable("LINK_TXT", $lng->txt('rep_robj_xmum_frm_list_use_grade'));
+                        $this->tpl->setVariable("VAL_DATE", substr($xGrade->timestamp, 0, 10) . " - " . substr($xGrade->timestamp, 11, 5));
+                        $this->tpl->setCurrentBlock("tbl_content");
+                        $this->tpl->parseCurrentBlock();
+                    }
+                }
+            }
+        }
+
+        $this->enable('header');
+        $this->enable('sort');
+        $this->setEnableHeader(true);
+    }
+
+    public static function overrideGrade($parentObj)
+    {
+        global $ilDB, $DIC, $lng;
+        $percentage = $_GET['newGrade'];
+        $DIC->database()->update(
+            'ut_lp_marks',
+            array(
+                "status_changed" => array('text', date("Y-m-d H:i:s", $_GET['timestamp'])),
+                "mark" => array('int', $percentage),
+            ),
+            array(
+                'obj_id' => array('int', $parentObj->object->getId()),
+                'usr_id' => array('int', $_GET['user_id']),
+            )
+        );
+        require_once('./Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask/classes/class.ilMumieTaskGradeSync.php');
+        require_once('Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask/classes/class.ilMumieTaskIdHashingService.php');
+        $hashed_user = ilMumieTaskIdHashingService::getHashForUser($_GET["user_id"], $parentObj->object);
+        $gradesync  = new ilMumieTaskGradeSync($parentObj->object, false);
+        if (!$gradesync->wasGradeOverriden($_GET["user_id"])) {
+            $ilDB->insert(
+                "xmum_grade_override",
+                array(
+                    'task_id' => array('integer', $parentObj->object->getId()),
+                    'usr_id' => array('text', $hashed_user),
+                    'new_grade' => array('integer', $percentage)
+                )
+            );
+        } else {
+            $ilDB->update(
+                "xmum_grade_override",
+                array(
+                    'new_grade' => array('integer', $percentage)
+                ),
+                array(
+                    'task_id' => array('integer', $parentObj->object->getId()),
+                    'usr_id' => array('text', $hashed_user),
+                )
+            );
```

Comment:
This should be extracted into a new function

Reply:
done